### PR TITLE
Fixing and documenting the CrowdGuard workflow tutorial

### PR DESCRIPTION
**Pull request by teoparvanov**
In this PR I've focused on enabling the CrowdGuard example to run on CPU (5 train/defend rounds are runnable within a reasonable amount of time). It is noteworthy, however, that this example cannot execute on a normal PC, as it requires at least 18GB of dedicated RAM. Running CrowdGuard on GPU-accelerated collaborators probably doesn't make much sense anyway as GPUs offer limited (if any) support for confidential computing.

Thanks, @bandatarunkumar for helping me thoroughly test the proposed changes!

PS: I've also added a step-by-step guide on how to run the `cifar10_crowdguard.py` demo script.

### CODE REVIEW

**Review by rahulga1 — APPROVED** [review]

File: openfl-tutorials/experimental/workflow/CrowdGuard/PoisoningAttackDemo.ipynb [hunk 1]
```diff
@@ -430,9 +430,8 @@
     "        state_dicts = [model.state_dict() for model in models]\n",
     "        state_dict = new_model.state_dict()\n",
     "        for key in models[1].state_dict():\n",
-    "            state_dict[key] = np.sum(\n",
```
Comment (bandatarunkumar) [inline openfl-tutorials/experimental/workflow/CrowdGuard/PoisoningAttackDemo.ipynb:433]: Hi Teo, can you please make the same changes about converting numpy to torch, using cpu for collaberator and changing affinity to metric in PoisoningAttackDemoReduced.ipynb?

Comment (teoparvanov) [reply openfl-tutorials/experimental/workflow/CrowdGuard/PoisoningAttackDemo.ipynb:433]: I couldn't find any instances of those issues in `PoisoningAttackDemoReduced.ipynb`, but I might also be missing something... Could you let me know of to the exact lines of code you're referring to?

Comment (bandatarunkumar) [reply openfl-tutorials/experimental/workflow/CrowdGuard/PoisoningAttackDemo.ipynb:433]: i tested it again with fresh clone with new python virtual env, i see that no change is required in PoisoningAttackDemoReduced.ipynb, please ignore this comment.

**Review by bandatarunkumar — APPROVED** [review]

**Review by kminhta — APPROVED** [review]
Great job updating the CrowdGuard example @teoparvanov !
